tests/test_cli.py

```diff
@@ -15,162 +17,438 @@ def cli_runner():
-    def test_cli_works(self, cli_runner):
+    def test_cli_no_options(self, cli_runner):
         result = cli_runner.invoke(cli.cli)
-        assert result.exit_code == 0
+        assert not result.output
         assert not result.exception
+        assert result.exit_code == 0
 
     def test_cli_with_version_option(self, cli_runner):
         result = cli_runner.invoke(cli.cli, ['--version'])
         assert not result.exception
         assert result.exit_code == 0
-        assert result.output.strip() == 'therapist, version {}'.format(__version__)
+        assert 'v{}'.format(__version__) in result.output
 
     def test_cli_with_help_option(self, cli_runner):
         result = cli_runner.invoke(cli.cli, ['--help'])
+        assert 'Usage:' in result.output
         assert not result.exception
         assert result.exit_code == 0
 
-    def test_cli_install(self, cli_runner, tmpdir):
+
+class TestInstall(object):
+    def test_install(self, cli_runner, tmpdir):
         p = Project(tmpdir.strpath)
         with chdir(p.path):
             result = cli_runner.invoke(cli.install)
+        assert 'Installing pre-commit hook...' in result.output
         assert not result.exception
         assert result.exit_code == 0
+        assert p.exists('.git/hooks/pre-commit')
+
+    def test_outside_repo(self, cli_runner, tmpdir):
+        with chdir(tmpdir.strpath):
+            result = cli_runner.invoke(cli.install)
+        assert 'Unable to locate git repo.' in result.output
+        assert result.exception
+        assert result.exit_code == 1
 
-    def test_cli_reinstall(self, cli_runner, tmpdir):
+    def test_try_reinstall(self, cli_runner, tmpdir):
         p = Project(tmpdir.strpath)
         with chdir(p.path):
             cli_runner.invoke(cli.install)
+            assert p.exists('.git/hooks/pre-commit')
+
+            result = cli_runner.invoke(cli.install)
+            assert 'The pre-commit hook has already been installed.' in result.output
+            assert not result.exception
+            assert result.exit_code == 0
+
+    def test_update_outdated(self, cli_runner, tmpdir):
+        p = Project(tmpdir.strpath)
+        with chdir(p.path):
+            cli_runner.invoke(cli.install)
+            assert p.exists('.git/hooks/pre-commit')
+
+            hook = p.read('.git/hooks/pre-commit')
+            hook_hash = cli.get_hook_hash(p.abspath('.git/hooks/pre-commit'))
+
+            p.write('.git/hooks/pre-commit', hook.replace('# THERAPIST {}'.format(hook_hash), '# THERAPIST n0tth3h45h'))
 
             result = cli_runner.invoke(cli.install, input='y')
+            assert 'You are not using the current version of the pre-commit hook.' in result.output
             assert not result.exception
             assert result.exit_code == 0
 
+            hook = p.read('.git/hooks/pre-commit')
+            assert '# THERAPIST {}'.format(hook_hash) in hook
+
+    def test_dont_update_outdated(self, cli_runner, tmpdir):
```

Comment:
yeah at the time i was coding this i thought maybe there would be a scenario where you wouldn't want to install to replace the hook. but when i think about it now, i can't think of a good reason why that would happen.